nvflare/app_common/aggregators/dxo_aggregator.py

```diff
@@ -0,0 +1,153 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvflare.apis.dxo import DXO, DataKind, MetaKey
+from nvflare.apis.fl_component import FLComponent
+from nvflare.apis.fl_context import FLContext
+from nvflare.app_common.aggregators.weighted_aggregation_helper import WeightedAggregationHelper
+from nvflare.app_common.app_constant import AppConstants
+
+
+class DXOAggregator(FLComponent):
+    def __init__(
+        self, exclude_vars=None, aggregation_weights=None, expected_data_kind: DataKind = DataKind.WEIGHT_DIFF
+    ):
+        """Perform accumulated weighted aggregation for one kind of corresponding DXO from contributors
+
+        Args:
+            exclude_vars ([type], optional): regex to match excluded vars during aggregation. Defaults to None.
+            aggregation_weights ([type], optional): dictionary to map contributor name to its aggregation weights.
+                                Defaults to None.
+            expected_data_kind: Expected DataKind for this kind of DXO
+        """
+        super().__init__()
+        self.expected_data_kind = expected_data_kind
+        self.aggregation_weights = aggregation_weights or {}
+        self.logger.debug(f"aggregation weights control: {aggregation_weights}")
+
+        self.aggregation_helper = WeightedAggregationHelper(exclude_vars=exclude_vars)
+        self.reset_aggregation_helper()
```

Comment:
Probably no need to call the reset_aggregation_helper here since the helper is just created.

Reply:
Good point. Removed.